receiver/sapnetweaverreceiver/metrics.go

```diff
@@ -40,86 +40,160 @@ var (
+			case models.StateColorGreen:
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 0, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateGrey)
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 1, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateGreen)
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 0, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateYellow)
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 0, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateRed)
+			case models.StateColorYellow:
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 0, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateGrey)
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 0, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateGreen)
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 1, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateYellow)
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 0, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateRed)
+			case models.StateColorRed:
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 0, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateGrey)
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 0, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateGreen)
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 0, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateYellow)
+				s.mb.RecordSapnetweaverSystemInstanceAvailabilityDataPoint(now, 1, response.Hostname, int64(response.InstanceNr), feature, metadata.AttributeControlStateRed)
+			default:
+				errs.AddPartial(1, fmt.Errorf("invalid control state color value for %s: %s", metricName, response.Dispstatus))
+			}
+		}
+	}
+}
+
+func (s *sapNetweaverScraper) recordSapnetweaverProcessAvailabilityDataPoint(now pcommon.Timestamp, processListResponse *models.GetProcessListResponse, errs *scrapererror.ScrapeErrors) {
+	metricName := "Process Availability"
+	if processListResponse.Process == nil {
+		err := formatErrorMsg(metricName, "", errValueNotFound)
+		errs.AddPartial(1, err)
+		return
+	}
+
+	for _, response := range processListResponse.Process.Item {
+		switch models.StateColor(*response.Dispstatus) {
+		case models.StateColorGray:
+			s.mb.RecordSapnetweaverProcessAvailabilityDataPoint(now, 1, response.Name, response.Description, metadata.AttributeControlStateGrey)
```

Comment:
This is a bit robust. Can we have 4 color variables and just set the corresponding one in the switch. Then set the metrics at the end?

```go


for _, response := range processListResponse.Process.Item {
     var gray, green, yellow, red int
     switch models.StateColor(*response.Dispstatus) {
     case models.StateColorGray:
          gray = 1
     case models.StateColorGreen:
          green = 1
     case models.StateColorYellow:
          yellow = 1
     case models.StateColorRed:
           red = 1
     }
     s.mb.RecordSapnetweaverProcessAvailabilityDataPoint(now, gray, response.Name, response.Description, metadata.AttributeControlStateGrey)
     s.mb.RecordSapnetweaverProcessAvailabilityDataPoint(now, green, response.Name, response.Description, metadata.AttributeControlStateGreen)
     s.mb.RecordSapnetweaverProcessAvailabilityDataPoint(now, yellow, response.Name, response.Description, metadata.AttributeControlStateYellow)
     s.mb.RecordSapnetweaverProcessAvailabilityDataPoint(now, red, response.Name, response.Description, metadata.AttributeControlStateRed)
}

```

Reply:
I like that a lot more thanks.